SimPEG/maps.py

```diff
@@ -616,6 +616,66 @@ def deriv(self, m, v=None):
         return self.sphericalDeriv(m)
 
 
+class VectorAmplitude(IdentityMap):
+    """
+    A vector amplitude map defined as
+
+    .. math::
+
+        v^2 = (v_x^2 + v_y^2 + v_z^2)
+
+    where :math:`v_x`, :math:`v_y` and :math:`v_z` are the
+    vector components. The derivative ...
+    """
+
+    def __init__(self, mesh=None, nP=None, **kwargs):
+
+        super().__init__(mesh, nP, **kwargs)
+
+        self.wires = Wires(("x", nP), ("y", nP), ("z", nP))
+        self.model = None
+
+    @property
+    def shape(self):
+        """
+        Shape of the matrix operation (number of indices x nP)
+        """
+        return (self.nP, 3*self.nP)
+
+    def _transform(self, model):
+        """
+        :param model:
+        :return:
+        """
+        return np.linalg.norm(self.wires*model, axis=0)
```

Comment:
Have you looked at all at using the amplitude squared rather than amplitude? I appreciate it is a different function, but derivatives are much nicer

`np.sum((self.wires*model)**2, axis=0)`

Reply:
So the square is technically happening in the reg.__call__. That's the confusing part. We are sort of mapping the parameters, but with the intention of using it in an L2 function? Or are you suggesting to minimize \| a^2 \|?

Reply:
here, I am thinking about minimizing | a^2 | (or | a^2 - aref^2| ). It would get rid of the square root and having to deal with a denominator (that could potentially go to zero).

Reply:
yep gotcha, haven't tried that.